File: Tyche.Android/account/src/main/java/com/felipearpa/tyche/account/login/LoginCredentialModel.kt
package com.felipearpa.tyche.account.login

import com.felipearpa.tyche.session.type.Password
import com.felipearpa.tyche.core.type.Email

data class LoginCredentialModel(
    val email: String,
    val password: String
)

fun LoginCredentialModel.isValid() = Email.isValid(this.email) && Password.isValid(this.password)

File: Tyche.iOS/Account/Sources/Account/Login/LoginCredentialModel.swift
import Foundation
import Session

struct LoginCredentialModel: Equatable {
    var username: String
    var password: String
}

extension LoginCredentialModel {
    func isValid() -> Bool {
        return Username.isValid(value: self.username)
        && Password.isValid(value: self.password)
    }
}
